malwarefront/src/commons/auth/actions.js

```diff
@@ -3,21 +3,32 @@ import authService from "./service";
 
 import { push } from "connected-react-router";
 
-function login(username, password, navPath) {
+function login(username, password, prevLocation) {
     return dispatch => {
         dispatch({ type: authActionTypes.LOGIN_REQUEST, username })
         authService.login(username, password)
             .then(user => {
-                dispatch({ type: authActionTypes.LOGIN_SUCCESS, user: user, navPath: null });
-                dispatch(push(navPath || "/"));
+                dispatch({ type: authActionTypes.LOGIN_SUCCESS, user: user, prevLocation: null });
+                dispatch(push(prevLocation || "/"));
+            })
+            .catch(err => {
+                // Error is unclonable and we can't directly put it in state
```

Comment:
suggestion, because i was wondering for solid 10 seconds `in what state` for some reason

```suggestion
                // Error is unclonable and we can't directly put it in react's state
```

Reply:
Actually it's not React state but location state (react-history uses https://developer.mozilla.org/en-US/docs/Web/API/History/pushState internally). We can't put whole Response object in state because it's unserializable (throws `DataCloneError: The object could not be cloned.`)